uiowa/uiowa
# Update uiowa_alerts to handle both versions of emergency website endpoint

Resolves https://github.com/uiowa/uiowa/issues/7687




<img width="1009" alt="Screenshot 2024-05-29 at 11 58 59 AM" src="https://github.com/uiowa/uiowa/assets/4663676/18044855-d61d-47c3-b8e1-f9de0502a8fc">


# How to test



```
ddev blt frontend && ddev blt ds --site=default && ddev drush @default.local cset uiowa_alerts.settings hawk_alert.source https://emergency.dev.drupal.uiowa.edu/api/active.json
```
See the old test alerts from emergency.dev.drupal.uiowa.edu show. This has remained unchanged but there is now a bit of code to look for the .json file extension on the endpoint. If it doesn't exist, it will load the new library.

```
ddev drush @default.local cset uiowa_alerts.settings hawk_alert.source https://emergency.stage.drupal.uiowa.edu/api/active && ddev drush @default.local cr
```
When the time comes we can update hook and set this new endpoint, clearing necessary caches and updating the default config for the source. We can also remove the older library and the code that handles the switch out.

See the test alert from emergency.stage.drupal.uiowa.edu show with the correct CT date time and a proper link to the website.

# Follow up

- Update hook to set source on existing, flush cached block.
- Default config change for new sites.
- Remove library switch logic, old library entry, old library js.
